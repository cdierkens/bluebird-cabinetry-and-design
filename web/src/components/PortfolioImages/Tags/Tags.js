import PropTypes from "prop-types";
import React from "react";
import Tag from "../Tag/Tag";

const Tags = ({ tags }) => (
  <>
    <span className="block text-blue-dark text-base ml-1 mb-1">Tags</span>
    <div className="p-1 shadow-md flex flex-wrap justify-start">
      {tags.map(({ value, checked, onChange }) => (
        <Tag
          value={value}
          checked={checked}
          onChange={onChange}
          label={value}
          key={value}
        />
      ))}
    </div>
  </>
);

Tags.propTypes = {
  tags: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      checked: PropTypes.bool.isRequired,
      onChange: PropTypes.func.isRequired,
    })
  ).isRequired,
};

export default Tags;
