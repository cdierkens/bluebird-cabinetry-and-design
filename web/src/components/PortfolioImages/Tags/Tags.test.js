import { render } from "@testing-library/react";
import React from "react";
import Tags from "./Tags";

it("matches snapshot", () => {
  const { asFragment } = render(<Tags tags={[]} />);

  expect(asFragment()).toMatchSnapshot();
});
