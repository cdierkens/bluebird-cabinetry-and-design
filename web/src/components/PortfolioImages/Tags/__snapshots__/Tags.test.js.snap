// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="block text-blue-dark text-base ml-1 mb-1"
  >
    Tags
  </span>
  <div
    class="p-1 shadow-md flex flex-wrap justify-start"
  />
</DocumentFragment>
`;
